scripts/bin/opx-config-fanout

```diff
@@ -17,173 +17,223 @@ import cps
-        # add ports to a transaction
-        for phy_port in phy_port_list:
-            ifobj = nas_if.make_interface_from_phy_port(phy_port)
-            print "Creating interface " + ifobj.get_attr_data(_g_if_name)
-            l.append(('rpc', ifobj.get()))
-            ifnames.append(ifobj.get_attr_data(_g_if_name))
-            target_number_of_ports -= 1
-            if target_number_of_ports == 0:
-                break
-
-        # do the actual port create
-        tr = cps_utils.CPSTransaction(l)
-
-        # commit
-        if tr.commit() == False:
-            print "Failed to create interfaces for breakout operation"
-        else:
-            print "Successfully created interfaces..."
-
-        # init qos default
-        base_qos_init.init_interfaces(ifnames)
-
+            print "Completed.............."
+
+            # look at thephysical port and determine the number of hardware ports
+            # that exist
+            target_number_of_ports = len(
+                    fp_obj.get()['data']['base-if-phy/front-panel-port/port'])
+            # if the mode is breakin... then need to create one port only
+            print "mode is",_mode
+            if _mode == 4:  # 1x1
+                target_number_of_ports = 1
+            elif _mode == 3: # 2x1
+                target_number_of_ports = 2
+            elif _mode == 7: # 4x4 
+                target_number_of_ports = 1 
+            else: # 4x1
+                target_number_of_ports = 4
+
+            phy_port_list = []
+            while len(phy_port_list) < target_number_of_ports:
+            # wait for all of the physical ports to appear - depends on timing
+                time.sleep(1)
+                phy_port_list = nas_if.physical_ports_for_front_panel_port(fp_obj)
+                print "len list",len(phy_port_list)
+                print "phy_port_list",phy_port_list
+
+            l = []
+            ifnames = []
+            # add ports to a transaction
+            for phy_port in phy_port_list:
+                ifobj = nas_if.make_interface_from_phy_port(phy_port)
+                print "Creating interface " + ifobj.get_attr_data(_g_if_name)
+                l.append(('rpc', ifobj.get()))
+                ifnames.append(ifobj.get_attr_data(_g_if_name))
+                target_number_of_ports -= 1
+                if target_number_of_ports == 0:
+                    break
+
+            # do the actual port create
+            tr = cps_utils.CPSTransaction(l)
+
+            # commit
+            if tr.commit() == False:
+                print "Failed to create interfaces for breakout operation"
+            else:
+                print "Successfully created interfaces..."
+
+            # init qos default
+            base_qos_init.init_interfaces(ifnames)
+    if pg_id:
+        print "change port-group breakout mode and port speed configuration"
+        call(["nas_port_group_cmd.py", '-s', pg_id, '-b', br_mode, '-S', port_speed])
```

Comment:
Doing this will tie the functionality of `opx-config-fanout` to `nas_port_group_cmd.py`. For now, that is ok, but in the future we should decouple them.